website/client/components/guilds/groupFormModal.vue

```diff
@@ -0,0 +1,324 @@
+import bFormCheckbox from 'bootstrap-vue/lib/components/form-checkbox';
+import bFormSelect from 'bootstrap-vue/lib/components/form-select';
+import bTooltip from 'bootstrap-vue/lib/components/tooltip';
+
+export default {
+  components: {
+    bModal,
+    bBtn,
+    bFormInput,
+    bFormCheckbox,
+    bFormSelect,
+    bTooltip,
+  },
+  data () {
+    let data = {
+      newGuild: {
+        id: '',
+        name: '',
+        type: 'guild',
+        privacy: 'private',
+        description: '',
+        guildInformation: '',
+        categories: [],
+        onlyLeaderCreatesChallenges: true,
+        guildLeaderCantBeMessaged: true,
+        privateGuild: true,
+        allowGuildInvationsFromNonMembers: true,
+        newLeader: '',
+      },
+      categoryOptions: [
+        {
+          label: 'animals',
+          key: 'animals',
+        },
+        {
+          label: 'artDesign',
+          key: 'art_design',
+        },
+        {
+          label: 'booksWriting',
+          key: 'books_writing',
+        },
+        {
+          label: 'comicsHobbies',
+          key: 'comics_hobbies',
+        },
+        {
+          label: 'diyCrafts',
+          key: 'diy_crafts',
+        },
+        {
+          label: 'education',
+          key: 'education',
+        },
+        {
+          label: 'foodCooking',
+          key: 'food_cooking',
+        },
+        {
+          label: 'healthFitness',
+          key: 'health_fitness',
+        },
+        {
+          label: 'music',
+          key: 'music',
+        },
+        {
+          label: 'relationship',
+          key: 'relationship',
+        },
+        {
+          label: 'scienceTech',
+          key: 'science_tech ',
+        },
+      ],
+      showCategorySelect: false,
+      members: ['one', 'two'],
+    };
+
+    let hashedCategories = {};
+    data.categoryOptions.forEach((category) => {
+      hashedCategories[category.key] = category.label;
+    });
+    data.categoriesHashByKey = hashedCategories;
+
+    return data;
+  },
+  mounted () {
+    let vm = this;
+    this.$root.$on('shown::modal', () => {
+      let editingGroup = vm.$store.state.editingGroup;
+      if (!editingGroup) return;
+      vm.newGuild.name = editingGroup.name;
+      vm.newGuild.type = editingGroup.type;
+      vm.newGuild.privacy = editingGroup.privacy;
+      if (editingGroup.description) vm.newGuild.description = editingGroup.description;
+      vm.newGuild.id = editingGroup._id;
+    });
+  },
+  computed: {
+    charactersRemaining () {
+      return 500 - this.newGuild.description.length;
+    },
+    title () {
+      if (!this.newGuild.id) return this.$t('createGuild');
+      return this.$t('updateGuild');
+    },
+  },
+  methods: {
+    toggleCategorySelect () {
+      this.showCategorySelect = !this.showCategorySelect;
+    },
+    async submit () {
+      if (this.$store.state.user.data.balance < 1 && !this.newGuild.id) {
+        alert('Not enough gems');
```

Comment:
can you add a TODO about updating this alerts with proper notifications once we have them?